src/modules/claims/claims.service.ts

```diff
@@ -1403,22 +1408,102 @@ export class ClaimsService {
     );
   }
 
+  /**
+   * Verifies:
+   * - That credential proof is valid
+   * - That credential was issued by authorized issuer
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return Boolean indicating if verified and array of error messages
+   */
+  async verifyVc(
+    vc: VerifiableCredential<RoleCredentialSubject>
+  ): Promise<CredentialVerificationResult> {
+    const errors: string[] = [];
+    const issuerDID = this._signerService.did;
+    const proofVerified = await this._verifiableCredentialService.verify(vc)
+    if (!proofVerified) {
+      errors.push(ERROR_MESSAGES.PROOF_NOT_VERIFIED);
+    }
+    const role = vc.credentialSubject.role.namespace;
+    let issuerVerified = true;
+    try {
+      await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
+    } catch(e) {
+      issuerVerified = false;
+      errors.push((e as Error).message)
+    }
+    return {
+      errors,
+      isVerified: proofVerified && issuerVerified
+    }
+  }
+
+    /**
+   * Verifies:
+   * - That off-chain claim was issued by authorized issuer
+   * - That off-chain claim proof is valid
+   *
+   * @param subjectDID The DID to try to resolve a credential for
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc") 
+   * @return Boolean indicating if verified and array of error messages
+   */
+     async verifyOffChainClaim(subjectDID: string, roleNamespace: string): Promise<CredentialVerificationResult> {
```

Comment:
IMO, this should accept the `OffChainClaim` interface of ew-credentials [https://github.com/energywebfoundation/ew-credentials/blob/55b84926b85a7e2261fffad[…]6/packages/vc-verification/src/models/verifiable-credentials.ts](https://github.com/energywebfoundation/ew-credentials/blob/55b84926b85a7e2261fffad8231821736d5febb6/packages/vc-verification/src/models/verifiable-credentials.ts#L12)

The idea is that there should be some symmetry between `async verifyVc(vc: VerifiableCredential<RoleCredentialSubject>): Promise<CredentialVerificationResult>` (which accepts the actual `VerifiableCredential`) and this `verifyOffChainClaim` method (which will accept an actual `offChainClaim`)

```suggestion
     async verifyOffChainClaim(offChainClaim: OffChainClaim): Promise<CredentialVerificationResult> {
```

Reply:
Sure, can do, but the two methods used for offchain verification only use the subject did and role name. How can I resolve those in ICL from the claim?